tantar/src/main/java/org/opencadc/tantar/BucketValidator.java

```diff
@@ -543,6 +544,47 @@ public void replaceArtifact(final Artifact artifact, final StorageMetadata stora
         }
     }
 
+    /**
+     * Update the values of the given Artifact with those from the given StorageMetadata.  This differs from a replace
+     * as it will not delete the original Artifact first, but rather update the values and issue a PUT.
+     *
+     * @param artifact        The Artifact to update.
+     * @param storageMetadata The StorageMetadata from which to update the Artifact's fields.
+     * @throws Exception Any unexpected error.
+     */
+    @Override
+    public void updateArtifact(final Artifact artifact, final StorageMetadata storageMetadata) throws Exception {
+        if (canTakeAction()) {
+            final TransactionManager transactionManager = artifactDAO.getTransactionManager();
+
+            try {
+                LOGGER.debug("Start transaction.");
+                transactionManager.startTransaction();
+
+                // By reusing the Artifact instance's ID we can have the original Artifact but wipe out the mutable
+                // fields below.
+                artifact.contentEncoding = storageMetadata.contentEncoding;
+                artifact.contentType = storageMetadata.contentType;
+                artifact.storageLocation = storageMetadata.getStorageLocation();
+
+                artifactDAO.put(artifact, true);
```

Comment:
In this case you can do a plain put(artifact) since you know the storageLocation is equal/correct and are only updating artifact metadata... that means the DAO computes checksum and only updates lastModified if the checksum changes. I think that is correct here. 

In fact, it further illustrates that **this** policy doesn't really need to check anything in this case: it could assign metadata values and do the put for every artifact and not have to replicate checking for differences...

Reply:
I modified the `put(Artifact)` call.